packages/flutter/lib/src/widgets/heroes.dart

```diff
@@ -998,7 +1001,32 @@ class HeroController extends NavigatorObserver {
     BuildContext toHeroContext,
   ) {
     final Hero toHero = toHeroContext.widget as Hero;
-    return toHero.child;
+    if (MediaQuery.maybeOf(toHeroContext) == null && MediaQuery.maybeOf(fromHeroContext) == null) {
```

Comment:
I understand that `MediaQueryData.padding` is the cause for the issue that this PR's fixing. But it feels a little too "particular" to me. Can there be other "contextual" state like `MediaQuery.padding` that needs this kind of transition? Is `MediaQueryData.padding` the special one?

Reply:
I agree that it feels particular and was thinking of doing what you suggested but it seems that padding is the only attribute with reported issues on this behavior since there are many Layout elements that rely on padding but may have different padding values based on the MediaQuery of the context. I was going to try to make a transition between the MediaQueryData as a whole but there's no way to lerp between two whole MediaQueryData objects (maybe this behavior can be added in MediaQuery?). Otherwise, the addition is not very costly but fixes many issues that could arise similar to the original issue so it is worth adding. But if you have any further suggestions please let me know.

Reply:
Unfortunately I don't think we'll be able to cover everything since people can introduce their own InheritedWidgets. Developers will have to provide their own flightShuttleBuilder implementation if a non-framework InheritedWidget is causing jumpy animation. According to https://master-api.flutter.dev/flutter/widgets/MediaQueryData-class.html there're a few other things other than padding that can cause problem but difference in padding seems to be the most common issue.

Reply:
I see.

I'm watching the video tutorial of `padding` and `viewPadding`. It seems that `padding.bottom` drops to 0 when a keyboard is raised. How does that affect our interpolation here? Can you try a `Hero` placed at the bottom of the screen and see what happens when the keyboard is raised?

Reply:
I tried adding a hero to the bottom of the page and the keyboard being raised does not cause an error with the interpolation. Since the padding.bottom drops to 0, the transition just interpolates between 0 and 0. Either combination of padding changes would not cause the transition to fail since the interpolation analyzes both MediaQuerys. Correct me if I'm wrong though.